addons/l10n_it_edi/models/l10n_it_edi_import.py

```diff
@@ -0,0 +1,346 @@
+
+        # Bank account. <2.4.2.13>
+        if move.move_type not in ('out_invoice', 'in_refund'):
+            if (acc_number := gettext(tree, './/DatiPagamento/DettaglioPagamento/IBAN')):
+                if move.partner_id and move.partner_id.commercial_partner_id:
+                    bank = self.env['res.partner.bank'].search([
+                        ('acc_number', '=', acc_number),
+                        ('partner_id', '=', move.partner_id.commercial_partner_id.id),
+                        ('company_id', 'in', [move.company_id.id, False])
+                    ], order='company_id', limit=1)
+                else:
+                    bank = self.env['res.partner.bank'].search([
+                        ('acc_number', '=', acc_number),
+                        ('company_id', 'in', [move.company_id.id, False])
+                    ], order='company_id', limit=1)
+                if bank:
+                    move.partner_bank_id = bank
+                else:
+                    message_to_log.append("%s<br/>%s" % (
+                        _("Bank account not found, useful informations from XML file:"),
+                        self._compose_info_message(
+                            tree, ['.//DatiPagamento//Beneficiario',
+                                './/DatiPagamento//IstitutoFinanziario',
+                                './/DatiPagamento//IBAN',
+                                './/DatiPagamento//ABI',
+                                './/DatiPagamento//CAB',
+                                './/DatiPagamento//BIC',
+                                './/DatiPagamento//ModalitaPagamento'])))
+        elif (elements := tree.xpath('.//DatiPagamento/DettaglioPagamento')):
+            message_to_log.append("%s<br/>%s" % (
+                _("Bank account not found, useful informations from XML file:"),
+                self._compose_info_message(tree, './/DatiPagamento')))
+
+        # Invoice lines. <2.2.1>
+        tag_name = './/DettaglioLinee' if not extra_info['simplified'] else './/DatiBeniServizi'
+        for element in tree.xpath(tag_name):
+            move_line_form = move.invoice_line_ids.create({
+                'move_id': move.id,
+                'tax_ids': [fields.Command.clear()]})
+            if move_line_form:
+                message_to_log += self._l10n_it_edi_import_line(element, move_line_form, extra_info)
+
+        # Global discount summarized in 1 amount
+        if (discount_elements := tree.xpath('.//DatiGeneraliDocumento/ScontoMaggiorazione')):
+            taxable_amount = float(move.tax_totals['amount_untaxed'])
+            discounted_amount = taxable_amount
+            for discount_element in discount_elements:
+                discount_sign = 1
+                if (discount_type := discount_element.xpath('.//Tipo') and discount_type[0].text == 'MG'):
+                    discount_sign = -1
+                if (discount_amount := gettext(discount_element, './/Importo')):
```

Comment:
you might be a big fan of :=, but is the minimum version for Odoo 16.4/17 already Python 3.8?